recipes/qt/5.x.x/conanfile.py

```diff
@@ -0,0 +1,710 @@
+            del self.settings.os.version
+
+    def configure(self):
+        #if self.settings.os != 'Linux':
+        #         self.options.with_libiconv = False # QTBUG-84708
+
+        if self.options.widgets and not self.options.gui:
+            raise ConanInvalidConfiguration("using option qt:widgets without option qt:gui is not possible. "
+                                            "You can either disable qt:widgets or enable qt:gui")
+        if not self.options.gui:
+            del self.options.opengl
+            del self.options.with_vulkan
+            del self.options.with_freetype
+            del self.options.with_fontconfig
+            del self.options.with_harfbuzz
+            del self.options.with_libjpeg
+            del self.options.with_libpng
+
+        if not self.options.qtmultimedia:
+            del self.options.with_libalsa
+            del self.options.with_openal
+
+        if self.options.qtwebengine:
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("Static builds of Qt Webengine are not supported")
+
+            if tools.cross_building(self.settings, skip_x64_x86=True):
+                raise ConanInvalidConfiguration("Cross compiling Qt WebEngine is not supported")
+
+            if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
+
+        if self.settings.os == "Android" and self.options.get_safe("opengl", "no") == "desktop":
+            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+
+        if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") == "dynamic":
+            raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
+
+        if self.options.get_safe("with_fontconfig", False) and not self.options.get_safe("with_freetype", False):
+            raise ConanInvalidConfiguration("with_fontconfig cannot be enabled if with_freetype is disabled.")
+
+        if self.options.multiconfiguration:
+            del self.settings.build_type
+
+        if not self.options.with_doubleconversion and str(self.settings.compiler.libcxx) != "libc++":
+            raise ConanInvalidConfiguration('Qt without libc++ needs qt:with_doubleconversion. '
+                                            'Either enable qt:with_doubleconversion or switch to libc++')
+
+        if tools.os_info.is_linux:
+            if self.options.qtwebengine:
+                self.options.with_fontconfig = True
+
+        if "MT" in self.settings.get_safe("compiler.runtime", default="") and self.options.shared:
+            raise ConanInvalidConfiguration("Qt cannot be built as shared library with static runtime")
+
+        config = configparser.ConfigParser()
+        config.read(os.path.join(self.recipe_folder, 'qtmodules%s.conf' % self.version))
```

Comment:
Is there a reason for using `.conf`? would have been easier to use another format?

Reply:
it's directly coming from upstream: https://code.qt.io/cgit/qt/qt5.git/tree/.gitmodules?h=v5.15.2

Reply:
Perfect!